src/components/DateInput/DateInput.tsx

```diff
@@ -60,21 +53,137 @@ const parseDate = (str: string, format: string): Date | undefined => {
  * <a href="/#/TextInput">TextInput</a> component (i.e. placeholder, etc.)
  *
  * */
-const DateInput = React.forwardRef<HTMLInputElement, DateInputProps>(function DateInput(
-  { format = 'MM/DD/YYYY', value, onChange, ...rest },
-  ref
-) {
+const DateInput: React.FC<DateInputProps> = ({
+  date,
+  format = 'MM/DD/YYYY',
+  withTime,
+  onChangeDate = noop,
+  ...rest
+}) => {
+  const dateFormatted = date ? dayjs(date) : dayjs();
+  const [currentMonth, setCurrentMonth] = useState(dateFormatted);
+  const [currentDate, setCurrentDate] = useState(date);
+  const [prevDate, setPrevDate] = useState(date);
+
+  const [open, setOpen] = useState(false);
+
+  const onNextMonth = useCallback(
+    e => {
+      e.preventDefault();
+      const next = currentMonth.add(1, 'month');
+      setCurrentMonth(next);
+    },
+    [currentMonth, setCurrentMonth]
+  );
+  const onPreviousMonth = useCallback(
+    e => {
+      e.preventDefault();
+      const next = currentMonth.subtract(1, 'month');
+      setCurrentMonth(next);
+    },
+    [currentMonth, setCurrentMonth]
+  );
+
+  const onCancel = useCallback(() => {
+    setCurrentDate(prevDate);
+    setOpen(false);
+  }, [setOpen, prevDate, setCurrentDate]);
+
+  const onApply = useCallback(
+    e => {
+      e.preventDefault();
+      setPrevDate(currentDate);
+      onChangeDate(currentDate);
+      setOpen(false);
+    },
+    [setOpen, setPrevDate, onChangeDate, currentDate]
+  );
+
+  const onExpand = useCallback(
+    e => {
+      e.preventDefault();
+      setOpen(true);
+    },
+    [setOpen]
+  );
+
+  const onDaySelect = useCallback(
+    dateChanged => {
+      const oldDate = dayjs(date);
+      const updated = dateChanged.hour(oldDate.hour()).minute(oldDate.minute());
+      // // @ts-ignore
```

Comment:
we can potentially remove  the `// @ts-ignore` if you do `const updated = .... as Date`;

Reply:
`// @ts-ignore` was eventually redundant 😓